core/tf_psa_crypto_common.h

```diff
@@ -208,6 +208,14 @@ static inline void mbedtls_xor(unsigned char *r,
         return;
     }
 #endif
+#if defined(MBEDTLS_COMPILER_IS_GCC) && __has_builtin(__builtin_constant_p)
```

Comment:
It looks like the CI is unhappy because `__has_builtin()` was added in [GCC 10](https://gcc.gnu.org/gcc-10/changes.html).

You should be able to use something like:
```c
#if defined(MBEDTLS_COMPILER_IS_GCC) && (__GNUC__ >= 10) && ...
```
To filter for GCC >= 10

Reply:
Looks like someone got here ahead of me -- `MBEDTLS_HAS_BUILTIN` is defined a bit further down in the file and looks like it will do the trick nicely; it explicitly checks if __has_builtin is available. I had to move that up in this file to make it available where necessary; I tried to move it among similar-looking macros.